lib/puma/dsl.rb

```diff
@@ -539,6 +539,22 @@ def lowlevel_error_handler(obj=nil, &block)
       @options[:lowlevel_error_handler] = obj
     end
 
+    # When force_shutdown_after is set, this allows you to control the message returned
+    # on a force shutdown to the client.
+    #
+    # @example
+    #   force_shutdown_error_response(500, {"Content-Type" => "application/json"}, [JSON.generate({message: "Server shutdown."})])
+    def force_shutdown_error_response(status, headers, response)
+      raise "Headers must be a hash" unless headers.is_a?(Hash)
```

Comment:
Same here. It just has to respond to each, it doesn't have to be a hash.

Reply:
Maybe `raise "Not a Rack-compatible response" unless [headers, response].all? {|x| x.respond_to?(:each)}"`

Reply:
Ah interesting, I didn't realize Rack solely cared about `each` for the headers too. I went with:

```ruby
      raise "Not a Rack-compatible header (must respond to each)" unless headers.respond_to?(:each)
      raise "Not a Rack-compatible response (must respond to each)" unless response.respond_to?(:each)
```

To err on the side of being a bit more specific/helpful on errors.